src/dosext/mfs/mfs.c

```diff
@@ -957,8 +958,11 @@ mfs_redirector(void)
 }
 
 int
-mfs_inte6(void)
+mfs_inte6(int fromDOS)
 {
+  if (fromDOS && WORD(REGS.ebx) == DOS_SUBHELPER_MFS_REDIR_INIT)
+    emufs_loaded = TRUE;
```

Comment:
I dismissed walking the Device Driver chain since the position in LOL is version specific. I'll rework as you said, using 0 helper.